util/stmtsummary/statement_summary.go

```diff
@@ -765,6 +772,10 @@ func (ssElement *stmtSummaryByDigestElement) add(sei *StmtExecInfo, intervalSeco
 		ssElement.execRetryCount += sei.ExecRetryCount
 		ssElement.execRetryTime += sei.ExecRetryTime
 	}
+
+	if sei.MergeStats != nil && sei.ExecStats != nil {
+		ssElement.totalExecStats = sei.MergeStats(ssElement.totalExecStats, sei.ExecStats)
+	}
```

Comment:
This part seems time-consuming, and it's inside a lock. This makes the lock contention of statements summary even more heavy. Did you do a benchmark when the concurrency is very high?